pyro/distributions/sine_bivariate_von_mises.py

```diff
@@ -0,0 +1,280 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import warnings
+from copy import copy
+from functools import reduce
+from math import pi
+
+import torch
+from torch.distributions import VonMises
+from torch.distributions.utils import broadcast_all
+
+from pyro.distributions import constraints
+from pyro.distributions.torch_distribution import TorchDistribution
+from pyro.distributions.util import broadcast_shape
+
+
+def _flat_dim(size):
+    return torch.prod(torch.tensor(size), dim=0)
+
+
+def log_I1(orders: int, value: torch.Tensor, terms=250):
```

Comment:
If you think this might be used elsewhere, you could move it to [pyro.ops.special](https://github.com/pyro-ppl/pyro/blob/dev/pyro/ops/special.py).

Reply:
It shows up in normalization constants for directional distribution (e.g. [Kent distribution](https://en.wikipedia.org/wiki/Kent_distribution)). I'll move it to `special`.